### PR TITLE
Upadated

### PR DESCRIPTION
**Pull request by datta513**

### CODE REVIEW

File: UPdate1.clj [hunk 1]
```diff
@@ -0,0 +1,46 @@
+(config
+ (text-field
+  :name "clientId" ; name is fixed
+  :label "Enter clientId"
+  :placeholder "")
```
Comment (datta513) [inline UPdate1.clj:5]: ihdaid

File: UPdate1.clj [hunk 2]
```diff
@@ -0,0 +1,46 @@
+(config
+ (text-field
+  :name "clientId" ; name is fixed
```
Comment (fivetran-vinodd) [inline UPdate1.clj]:
```suggestion
  :name "clientId" 
```

File: UPdate1.clj [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+  :label "Enter clientId"
+  :placeholder "")
+ (password-field
+  :name "clientSecret" ; name is fixed
```
Comment (fivetran-vinodd) [inline UPdate1.clj]:
```suggestion
  :name "clientSecret" 
```

File: UPdate1.clj (hunk 1 above)
Comment (fivetran-vinodd) [inline UPdate1.clj]:
```suggestion
  :placeholder "Enter your Client ID")
```

File: UPdate1.clj [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+  :label "Enter clientSecret"
+  :placeholder "")
```
Comment (fivetran-vinodd) [inline UPdate1.clj]:
```suggestion
  :placeholder "Enter Your Client Secret")
```

File: UPdate1.clj [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+ (oauth2/authorization-code-flow-client-credentials
```
Comment (fivetran-vinodd) [inline UPdate1.clj]: It's refresh-token with client credential not authorisation code flow

File: UPdate1.clj [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+    expires-in :<= "expires_in")))
+ 
+)
+(default-source
+ (http/get :base-url "https://api.paychex.com"
+           (header-params "Accept" "application/json")) 
+ (auth/oauth2)
+ (paging/no-pagination)
+ (error-handler
+  (when :status 401 :action refresh)
```
Comment (fivetran-vinodd) [inline UPdate1.clj]: add this  `(when :status 429 :action rate-limit)`  error handler whenever you implement oauth authentication

Comment (fivetran-vinodd) [reply UPdate1.clj]: remove 402 and 403 error handlers

File: UPdate1.clj [hunk 7]
```diff
@@ -0,0 +1,46 @@
...
+    (when :status 402 :action fail)
+    (when :status 403 :action fail)
+  ))
+    
+(entity COMPANY
+        (api-docs-url "https://developer.paychex.com/documentation#tag/Company")
+        (source (http/get :url "/companies")
+                (query-params "displayid" {"displayid"})
```
Comment (fivetran-vinodd) [inline UPdate1.clj]: this query param is not required please remove

File: UPdate1.clj [hunk 8]
```diff
@@ -0,0 +1,46 @@
...
+  :placeholder "")
+ (password-field
+  :name "clientSecret" ; name is fixed
+  :label "Enter clientSecret"
+  :placeholder "")
+ (oauth2/authorization-code-flow-client-credentials
+  (token
+   (source
+    (http/post
+     url:"https://api.paychex.com/auth/oauth/v2/token"
```
Comment (fivetran-vinodd) [inline UPdate1.clj]:
```suggestion
     url: "https://api.paychex.com/auth/oauth/v2/token"
```

File: UPdate1.clj [hunk 9]
```diff
@@ -0,0 +1,46 @@
...
+     (body-params
+      "response_type" "code"
```
Comment (fivetran-vinodd) [inline UPdate1.clj]: remove this it's not required

File: UPdate1.clj [hunk 10]
```diff
@@ -0,0 +1,46 @@
...
+      "client_id" "${CLIENT-ID}"
+      "client_secret" "${CLIENT-SECRET}"
+    ;;   "scope" "contact_read all_contact_read""
+    "grant_type " " authorization_code "
```
Comment (fivetran-vinodd) [inline UPdate1.clj]:
```suggestion
    "grant_type" "client_credentials"
```

**Pull request merged**